Library/Homebrew/extend/os/mac/formula_cellar_checks.rb

```diff
@@ -56,10 +58,41 @@ def check_python_framework_links(lib)
     EOS
   end
 
+  def check_linkage
+    return unless formula.prefix.directory?
+    keg = Keg.new(formula.prefix)
+    checker = LinkageChecker.new(keg, formula)
+    s = ""
+
+    if checker.broken_dylibs?
+      s += <<-EOS.undent
+        Broken dylib links found:
+          #{checker.broken_dylibs.to_a * "\n          "}
+
+        It is likely the installation was broken.
+        Please check formula and make sure it is built from source.
+      EOS
+    end
+
+    if checker.undeclared_deps?
+      s += <<-EOS.undent
+        Undeclared dependencies found:
+          #{checker.undeclared_deps * "\n          "}
+
+        Formula is required to declare all direct dependencies.
+        Please check whether these dependencies are needed.
+        If so, declare them in the formula using `depends_on`.
```

Comment:
This seems too tentative and conditional to me. If I "check whether these dependencies are needed" and decide "nope, they're not," the audit will nonetheless continue to fail until I either add the depends_on or take some affirmative steps to prevent the linkage from occurring.